app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -359,4 +386,86 @@ public void onMicrolocationsDownloadDone(MicrolocationDownloadEvent event) {
 
     }
 
-}
+    /*@Subscribe
+    public void showErrorDialog(HandleErrorDialogEvent event){
+        downloadProgress.setVisibility(View.GONE);
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setTitle(R.string.error)
+                .setMessage(event.errorType + ": " + event.errorDesc)
+                .setNeutralButton(R.string.ok, null)
+                .create();
+        builder.show();
+    }*/
+
+    @Subscribe
+    public void showNetworkDialog(ShowNetworkDialogEvent event){
+        downloadProgress.setVisibility(View.GONE);
+        AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
+        builder.setTitle(getString(R.string.net_unavailable))
+                .setMessage(getString(R.string.turn_on))
+                .setPositiveButton(R.string.action_settings, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        Intent setNetworkIntent = new Intent(Settings.ACTION_SETTINGS);
+                        startActivity(setNetworkIntent);
+                    }
+                })
+                .setNeutralButton(R.string.cancel, null)
+                .create();
+        builder.show();
+    }
+
+    @Subscribe
+    public void downloadData(DataDownloadEvent event){
+        DataDownload download = new DataDownload();
+        download.downloadVersions();
+        downloadProgress.setVisibility(View.VISIBLE);
+        Log.d("DataNotif", "Download has started");
+    }
+
+    @Subscribe
+    public void tempEvent(RetrofitError cause) {
+        ConnectivityManager connMgr = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo netinfo = connMgr.getActiveNetworkInfo();
+        switch(cause.getKind()){
+            case CONVERSION: {
+                Log.d(TYPE, "ConversionError");
+                errorType="Conversion Error";
+                errorDesc=String.valueOf(cause.getCause());
+                break;
+            }
+            case HTTP: {
+                Log.d(TYPE, "HTTPError");
+                errorType="HTTP Error";
+                errorDesc=String.valueOf(cause.getResponse().getStatus());
+                Log.d(ERROR_CODE, String.valueOf(cause.getResponse().getStatus()));
+                break;
+            }
+            case UNEXPECTED: {
+                Log.d(TYPE, "UnexpectedError");
+                errorType="Unexpected Error";
+                errorDesc=String.valueOf(cause.getCause());
+                break;
+            }
+            case NETWORK: {
+                if(!(netinfo!=null && netinfo.isConnected())){ //Network not connected error
+                    OpenEventApp.postEventOnUIThread(new ShowNetworkDialogEvent(MainActivity.this));
```

Comment:
Yes this is what was in my mind. But you can move the switch-case to a static function. like 
```java
Retrofit error = /* Obtained from somewhere */
errorType = NetworkUtils.getNetworkError(error)
if (errorType == null && !connected()) {
    // if network error dialog is not showing
    // show network connection dialog
} else {
    // display normal error dialog.
}
```